### PR TITLE
Add isStickyTask to poll request

### PR DESCRIPTION
**Pull request by vancexu**
related server change: https://github.com/uber/cadence/pull/510/files

### CODE REVIEW

File: idl/github.com/uber/cadence/shared.thrift [hunk 1]
```diff
@@ -735,6 +735,7 @@ struct PollForDecisionTaskRequest {
   10: optional string domain
   20: optional TaskList taskList
   30: optional string identity
+  40: optional bool isStickyTask
```
Comment (yiminc) [inline idl/github.com/uber/cadence/shared.thrift]: I thought we decided that we need an enum, not a boolean.

Comment (yiminc) [reply idl/github.com/uber/cadence/shared.thrift]: also, could it make sense to add the enum to the TaskList struct, instead of here?

Comment (vancexu) [reply idl/github.com/uber/cadence/shared.thrift]: I agree it make sense to add that info to TaskList, changed and tested

File: internal/internal_task_pollers.go [hunk 2]
```diff
@@ -242,27 +242,22 @@ func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
...
-
-func (wtp *workflowTaskPoller) release(isStickyTask bool) {
+func (wtp *workflowTaskPoller) release(kind s.TaskListKind) {
 	if wtp.disableStickyExecution {
 		return
 	}
 
 	wtp.requestLock.Lock()
-	if isStickyTask {
+	if kind.Equals(s.TaskListKindWorker) {
```
Comment (yiminc) [inline internal/internal_task_pollers.go]: use == to compare for enum type

File: internal/internal_task_pollers.go [hunk 3]
```diff
@@ -242,27 +242,22 @@ func (wtp *workflowTaskPoller) ProcessTask(task interface{}) error {
...
 	} else {
 		wtp.pendingRegularPollCount--
 	}
 	wtp.requestLock.Unlock()
 }
 
-func (wtp *workflowTaskPoller) updateBacklog(isStickyTask bool, backlogCountHint int64) {
-	if !isStickyTask || wtp.disableStickyExecution {
+func (wtp *workflowTaskPoller) updateBacklog(taskListKind s.TaskListKind, backlogCountHint int64) {
+	if taskListKind.Equals(s.TaskListKindNormal) || wtp.disableStickyExecution {
```
Comment (yiminc) [inline internal/internal_task_pollers.go]: use ==

File: idl/github.com/uber/cadence/shared.thrift [hunk 4]
```diff
@@ -196,6 +196,11 @@ enum HistoryEventFilterType {
   CLOSE_EVENT,
 }
 
+enum TaskListKind {
+  NORMAL,
+  WORKER,
```
Comment (yiminc) [inline idl/github.com/uber/cadence/shared.thrift]: rename WORKER to STICKY

File: internal/internal_task_pollers.go (hunk 2 above)
Comment (vancexu) [reply internal/internal_task_pollers.go]: changed, what's the benefits of using ==? Equals function itself use == inside.

**Pull request merged**